sensor-validate.c

```diff
@@ -1,28 +1,28 @@
 #include "sensor-validate.h"
 
-int _give_me_a_good_name(double value, double nextValue, double maxDelta) {
+int CheckSuddenJump(double value, double nextValue, double maxDelta) {
   if(nextValue - value > maxDelta) {
     return 0;
   }
   return 1;
 }
 
-int validateSOCreadings(double* values, int numOfValues) {
+int validateSOCreadings(double* values, int numOfValues, double maxDelta) {
   int lastButOneIndex = numOfValues - 1;
   for(int i = 0; i < lastButOneIndex; i++) {
-    if(!_give_me_a_good_name(values[i], values[i + 1], 0.05)) {
+    if(!CheckSuddenJump(values[i], values[i + 1], maxDelta)) {
       return 0;
     }
   }
   return 1;
 }
 
-int validateCurrentreadings(double* values, int numOfValues) {
+int validateCurrentreadings(double* values, int numOfValues, double maxDelta) {
   int lastButOneIndex = numOfValues - 1;
-  for(int i = 0; i < lastButOneIndex; i++) {
-    if(!_give_me_a_good_name(values[i], values[i + 1], 0.1)) {
-      return 0;
-    }
+  int returnvalue = 1;
+  if(values != 0)
+  {
+      returnvalue = validateSOCreadings(values,lastButOneIndex,maxDelta);
```

Comment:
Current and Soc are two different things.
But Current is using Soc here. That means changing the Soc implementation will inadvertently change the current logic as well.
Make an abstraction that can check any set of values and use that from both current and Soc